library/lua/gui.lua

```diff
@@ -885,6 +885,56 @@ ZScreenModal.ATTRS{
     pass_mouse_clicks = false,
 }
 
+-- DFHack textures
+--------------------------
+
+-- Preloaded DFHack Assets
+-- Use this handles if you need to get dfhack standard textures
+local texpos_handles = {
+    green_pin = dfhack.textures.loadTileset('hack/data/art/green-pin.png', 8, 12),
+    red_pin = dfhack.textures.loadTileset('hack/data/art/red-pin.png', 8, 12),
+    icons = dfhack.textures.loadTileset('hack/data/art/icons.png', 8, 12),
+    on_off = dfhack.textures.loadTileset('hack/data/art/on-off.png', 8, 12),
+    control_panel = dfhack.textures.loadTileset('hack/data/art/control-panel.png', 8, 12),
+    border_thin = dfhack.textures.loadTileset('hack/data/art/border-thin.png', 8, 12),
+    border_medium = dfhack.textures.loadTileset('hack/data/art/border-medium.png', 8, 12),
+    border_bold = dfhack.textures.loadTileset('hack/data/art/border-bold.png', 8, 12),
+    border_panel = dfhack.textures.loadTileset('hack/data/art/border-panel.png', 8, 12),
+    border_window = dfhack.textures.loadTileset('hack/data/art/border-window.png', 8, 12),
+}
+
+-- Methods to obtain valid texposes by handles
+function tp_green_pin(offset)
+    return dfhack.textures.getTexposByHandle(texpos_handles.green_pin[offset])
+end
+function tp_red_pin(offset)
+    return dfhack.textures.getTexposByHandle(texpos_handles.red_pin[offset])
+end
+function tp_icons(offset)
+    return dfhack.textures.getTexposByHandle(texpos_handles.icons[offset])
+end
+function tp_on_off(offset)
+    return dfhack.textures.getTexposByHandle(texpos_handles.on_off[offset])
+end
+function tp_control_panel(offset)
+    return dfhack.textures.getTexposByHandle(texpos_handles.control_panel[offset])
+end
+function tp_border_thin(offset)
+    return dfhack.textures.getTexposByHandle(texpos_handles.border_thin[offset])
+end
+function tp_border_medium(offset)
+    return dfhack.textures.getTexposByHandle(texpos_handles.border_medium[offset])
+end
+function tp_border_bold(offset)
+    return dfhack.textures.getTexposByHandle(texpos_handles.border_bold[offset])
+end
+function tp_border_panel(offset)
+    return dfhack.textures.getTexposByHandle(texpos_handles.border_panel[offset])
+end
+function tp_border_window(offset)
+    return dfhack.textures.getTexposByHandle(texpos_handles.border_window[offset])
+end
+
```

Comment:
Looking at the usage in `plugins/lua/buildingplan/pens.lua`, These feel a little out of place in this file. could you move them to a new `gui/textures.lua` file? `gui.lua` can add `local textures = require('gui.textures')`